Mathlib/CategoryTheory/Equivalence.lean

```diff
@@ -388,6 +388,22 @@ theorem cancel_counitInv_right_assoc' {W X X' Y Y' Z : D} (f : W ⟶ X) (g : X 
     f ≫ g ≫ h ≫ e.counitInv.app Z = f' ≫ g' ≫ h' ≫ e.counitInv.app Z ↔
     f ≫ g ≫ h = f' ≫ g' ≫ h' := by simp only [← Category.assoc, cancel_mono]
 
+/--
+If a functor `F` is an equivalence, then right-cancellation is possible in the sense that
+`G₁ ⋙ F ≅ G₂ ⋙ F` induces `G₁ ≅ G₂`.
+-/
+def isoCancelRight {E : Type*} [Category E] {G₁ G₂ : E ⥤ D} (F : D ≌ C)
+    (α : G₁ ⋙ F.functor ≅ G₂ ⋙ F.functor) : (G₁ ≅ G₂) :=
+  isoCancelRightOfWeaklyLeftInvertible F.unitIso α
```

Comment:
I am not sure we need these new definitions, because they can be obtained easily through the `Functor.FullyFaithful` API:
```suggestion
  F.congrRight.fullyFaithfulFunctor.preimageIso α
```

Reply:
Thanks for this great pointer! I will close this PR then.